core/src/main/java/cz/xtf/core/openshift/OpenShifts.java

```diff
@@ -183,22 +208,53 @@ private static String downloadOpenShiftBinary(String version) {
 		File ocTarFile = new File(workdir, "oc.tar.gz");
 		File ocFile = new File(workdir, "oc");
 
+		final String ocUrl = clientLocation + ocFileName;
 		try {
-			URL requestUrl = new URL(clientLocation + "oc.tar.gz");
-			FileUtils.copyURLToFile(requestUrl, ocTarFile, 20_000, 300_000);
+			URL requestUrl = new URL(ocUrl);
+
+			if (trustAll) {
+				Https.copyHttpsURLToFile(requestUrl, ocTarFile, 20_000, 300_000);
+			} else {
+				FileUtils.copyURLToFile(requestUrl, ocTarFile, 20_000, 300_000);
+			}
 
 			executeCommand("tar", "-xf", ocTarFile.getPath(), "-C", workdir.getPath());
 			FileUtils.deleteQuietly(ocTarFile);
 
 			return ocFile.getAbsolutePath();
 		} catch (IOException | InterruptedException e) {
-			throw new IllegalStateException("Failed to download and extract oc binary from " + clientLocation + "oc.tar.gz", e);
+			throw new IllegalStateException("Failed to download and extract oc binary from " + ocUrl, e);
 		}
 	}
 
 	public static String getVersion() {
-		String content = Https.httpsGetContent(OpenShiftConfig.url() + "/version/openshift");
-		return ModelNode.fromJSONString(content).get("gitVersion").asString().replaceAll("^v(.*)", "$1");
+		final String ocp3UrlVersion = OpenShiftConfig.url() + "/version/openshift";
+		if (Https.getCode(ocp3UrlVersion) == 200) { // for OCP 3
+			String content = Https.httpsGetContent(ocp3UrlVersion);
+			return ModelNode.fromJSONString(content).get("gitVersion").asString().replaceAll("^v(.*)", "$1");
+		} else { // for OCP version > 3
+			return apiResources("config.openshift.io/v1/clusterversions/version").get("status").get("desired").get("version").asString();
+		}
+	}
+
+	private static ModelNode apiResources(final String apiRequest) {
```

Comment:
Can method `apiResources(String apiRequest)` be moved to `OpenShift` as a general access method for api which is not covered otherwise. What about calling it in general way E.g. OpenShift.getResources(string resourceIdentifier)

Also can this be implemented in this way `Http.get(OpenShiftConfig.url() + "/apis/network.openshift.io/v1/hostsubnets").trustAll().bearerAuth(getMasterToken()).execute().response();` it seams shorter and readable